### PR TITLE
Detect and remove stuck queues

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: #110172

Based on: https://github.com/moderntribe/tribe-common/pull/736

This PR adds code, in the EA Imports Queue Processor, to detect and kill stuck queues.

mu-plugin to stuck the queues: https://gist.github.com/lucatume/32dba19eeb6497dd0fe578d5217c7d2b
video: http://p.tri.be/diC0nr

### CODE REVIEW

File: src/Tribe/Aggregator/Record/Queue.php [hunk 1]
```diff
@@ -439,5 +439,25 @@ protected function release_lock() {
...
+	 *
+	 * @return mixed
+	 */
+	public function is_stuck() {
+		return false;
+	}
+
+	/**
+	 * Orderly closes the queue process.
+	 *
```
Comment (bordoni) [inline src/Tribe/Aggregator/Record/Queue.php]: `@since`

File: src/Tribe/Aggregator/Record/Queue_Interface.php [hunk 2]
```diff
@@ -61,4 +61,18 @@ public function is_in_progress();
...
+	/**
+	 * Whether the current queue process is stuck or not.
+	 *
+	 * @return bool
+	 */
+	public function is_stuck();
+
+	/**
+	 * Orderly closes the queue process.
+	 *
```
Comment (bordoni) [inline src/Tribe/Aggregator/Record/Queue_Interface.php]: `@since`

File: src/Tribe/Aggregator/Record/Queue_Interface.php [hunk 3]
```diff
@@ -61,4 +61,18 @@ public function is_in_progress();
 	 * @return string
 	 */
 	public function get_queue_type();
+
+	/**
+	 * Whether the current queue process is stuck or not.
+	 *
```
Comment (bordoni) [inline src/Tribe/Aggregator/Record/Queue_Interface.php]: `@since`

File: src/Tribe/Aggregator/Record/Async_Queue.php [hunk 4]
```diff
@@ -307,4 +307,44 @@ public function get_queue_type() {
...
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function is_stuck() {
+		if ( ! empty( $this->record->meta['queue_id'] ) ) {
+			$queue_id = $this->record->meta['queue_id'];
+
+			return Tribe__Process__Queue::is_stuck( $queue_id );
```
Comment (barryhughes) [inline src/Tribe/Aggregator/Record/Async_Queue.php]: This line is a supremely readable little piece of code :smiley:

**Comment by lucatume** [comment]
I've updated the code to handle and show errors: https://central.tri.be/issues/110172

**Pull request merged**